daffodil-lib/build.sbt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GenJarLicense._
+
+additionalJarLicenseText := Some(
```

Comment:
Each submodule can have a build.sbt file added to provide submodule specific properties. This change defined a new property that submodule can set if they want additional license text specific to the submodule added to the generic Apache license that gets put in the jar. I was just rying to reduce dupliation, but I'll create an alternative pull request that just manages all the files manually and gets rid of the generator and see which is better. Manageing this additional licenses probably isn't significantly more effort, and is certainly more clear.

The problem with that plugin was that it only specifiied licenses of dependencies, and only if those dependencies included license information. Many didn't, or was incomplete, so we had to do a lot of manual effort to get the result correct. It just wasn't sufficient for Apache's licensing needs.

Additionally, that plugin didn't do anything for Apache license compatible code that we took and put in our source tree/jars. E.g. XML schemas, some sutff we took from scala. These aren't dependencies, we just copied code into our tree, so sbt doesn't know anythinga bout those. Those kinds of things are the things to go in these license since we actually distribute those. We don't distribute dependencies except for in your CLI package, which includes a separate license informatino for each dependency.

Reply:
Thanks for the reminder. I had forgotten about this stuff, but I can see how automated license management would require quite a bit more adherence to various standard ways to advertise one's license, and the world of licenses is just too chaotic out there for that.

Reply:
PR https://github.com/apache/incubator-daffodil/pull/360 is an alternative to this one that just manages the licenses manually and removes the complexity of the generator.